packages/ts-scripts/src/cdk/components/site.ts

```diff
@@ -0,0 +1,94 @@
+import { createCloudfront } from './cloudfront'
+import { createRoute } from './r53'
+import {
+  Stack,
+  aws_route53 as route53,
+  aws_cloudfront as cloudfront,
+  aws_route53_targets as targets,
+  aws_s3 as s3,
+  aws_s3_deployment as deploy,
+} from 'aws-cdk-lib'
+import { Certificate, ICertificate } from 'aws-cdk-lib/aws-certificatemanager'
+import { ACM } from 'aws-sdk'
+
+interface CreateSiteInterface {
+  domain: string
+  hostedDomainName?: string
+  defaultRootObject?: string
+  location: string
+}
+
+const findCert = async (stack: Stack, domain: string): Promise<ICertificate> => {
+  const acm = new ACM({
+    region: 'us-east-1',
+  })
+  const certDomain = `*.${domain}`
+  const result = await new Promise<ACM.ListCertificatesResponse>((resolve, reject) =>
+    acm.listCertificates({}, (error, data) => {
+      if (error) {
+        console.error(error)
+        reject(error)
+      }
+
+      return resolve(data)
+    }),
+  )
+
+  const domainCertificate = result.CertificateSummaryList?.find((cert) => cert.DomainName === certDomain)
+
+  if (!domainCertificate || !domainCertificate.CertificateArn) {
+    throw new Error('Certificate not found')
+  }
+
+  return Certificate.fromCertificateArn(stack, 'cert', domainCertificate?.CertificateArn)
+}
+
+export const createSite = async (
+  stack: Stack,
+  {
+    domain,
+    defaultRootObject = 'index.html',
+    hostedDomainName = 'dev.paas.reapit.cloud',
+    location,
+  }: CreateSiteInterface,
+) => {
+  const hostedZone = route53.HostedZone.fromLookup(stack, 'hosted-zone', { domainName: hostedDomainName })
+  const certificate = await findCert(stack, hostedDomainName)
+
+  const bucket = new s3.Bucket(stack, 'bucket', {
+    websiteIndexDocument: defaultRootObject,
+    websiteErrorDocument: defaultRootObject,
+    publicReadAccess: true,
+  })
+
+  const deploymentBucket = new deploy.BucketDeployment(stack, 'deployment', {
+    sources: [deploy.Source.asset(location)],
+    destinationBucket: bucket,
+  })
+
+  console.log('deployment', deploymentBucket.deployedBucket.bucketDomainName)
```

Comment:
still needed?

Reply:
Yea, sonar cloud won't let me ignore that I had a new class declaration without it being used

Reply:
it didn't like it X'D